File: csharp/csharp_fileutil/CommandEngine/AbstractCommand.cs
using System;
using System.Collections.Generic;
using System.Collections.ObjectModel;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace csharp_fileutil.CommandEngine
{
    public abstract class AbstractCommand
    {
        protected List<string> Arguments { get; private set; }
        private readonly IList<String> AvailableCommands = 
                                       new ReadOnlyCollection<string> (new List <string> { "usage", "exit" });

        public AbstractCommand(List<string> args)
        {
            Arguments = args;
        }

        protected string GetAvailableCommands()
        {
            return String.Join(", ", AvailableCommands);
        }

        public abstract void Execute();
    }
}

File: python/AbstractCommand.py
from abc import ABC, abstractmethod


class AbstractCommand(ABC):
    def __init__(self, args):
        self.arguments = []
        self.arguments += args[1:]

    @classmethod
    def _note_preface(cls):
        return str("NOTE: ")

    @property
    def _available_commands(self):
        return ["usage", "exit"]

    @abstractmethod
    def execute(self):
        pass

